codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/AWSSmokeTestGenerator.kt

```diff
@@ -0,0 +1,77 @@
+    }
+
+    override fun renderCustomFilePrivateVariables(writer: SwiftWriter) {
+        writer.write("fileprivate let regionFromEnv = ProcessInfo.processInfo.environment[\"AWS_SMOKE_TEST_REGION\"]")
+        writer.write("fileprivate let tagsToSkip = (ProcessInfo.processInfo.environment[\"AWS_SMOKE_TEST_SKIP_TAGS\"] ?? \"\").components(separatedBy: \",\")")
+    }
+
+    override fun handleVendorParams(vendorParams: ObjectNode, writer: SwiftWriter) {
+        val nameToValueMappings = getFormattedVendorParams(vendorParams)
+        nameToValueMappings.forEach { mapping ->
+            writer.write("config.${mapping.key} = ${mapping.value}")
+        }
+    }
+
+    // Converts trait definition vendor param key:value pairs to Swift SDK config field:value pairs.
+    private fun getFormattedVendorParams(vendorParams: ObjectNode): Map<String, String> {
+        val formattedMapping = mutableMapOf<String, String>()
+        vendorParams.members.forEach { originalMapping ->
+            when (originalMapping.key.value) {
+                /* BaseAwsVendorParams members */
+                "region" -> {
+                    // Take region value retrieved from environment variable if present; otherwise, take from trait definition.
+                    val regionValue = "regionFromEnv ?? \"${originalMapping.value.expectStringNode().value}\""
+                    formattedMapping.put("region", regionValue)
+                    formattedMapping.put("signingRegion", regionValue)
+                }
+                "sigv4aRegionSet" -> { /* no-op; setting multiple signing regions in config is unsupported atm. */ }
+                "uri" -> { formattedMapping.put("endpoint", "\"${originalMapping.value.expectStringNode().value}\"") }
+                "useFips" -> { formattedMapping.put("useFIPS", originalMapping.value.expectBooleanNode().value.toString()) }
+                "useDualstack" -> { formattedMapping.put("useDualStack", originalMapping.value.expectBooleanNode().value.toString()) }
+                "useAccountIdRouting" -> { /* no-op; setting account ID routing in config is unsupported atm. */ }
+
+                /* S3VendorParams members */
+                "useAccelerate" -> { formattedMapping.put("accelerate", originalMapping.value.expectBooleanNode().value.toString()) }
+                "useMultiRegionAccessPoints" -> {
+                    // Name for corresponding config in Swift SDK is: `disableMultiRegionAccessPoints`; value needs to be flipped.
+                    formattedMapping.put("disableMultiRegionAccessPoints", (!(originalMapping.value.expectBooleanNode().value)).toString())
+                }
+                "useGlobalEndpoint", "forcePathStyle", "useArnRegion" -> {
+                    // No change needed for these
+                    formattedMapping.put(originalMapping.key.value, originalMapping.value.expectBooleanNode().value.toString())
+                }
+            }
+        }
+        return formattedMapping
+    }
```

Comment:
This "translates" vendor params officially defined in Smithy vendor param shapes that service teams can use to specify client config values in their smoke tests. There're a couple that we don't have config bindings for yet.